### DISCUSSION TITLE
Text summarization task

### DISCUSSION THREAD

**Discussion by ramkumarkoppu** [keras-team/keras-hub#1079] [General] [answered]
If I want to do text summarization with GPT-2 model using keras_nlp, how do I do it? is there any API available for text summarization?

**Comment by jbischof** [comment]
@ramkumarkoppu yes, you can do this with any `CausalLM` model in our library. You can fine tune the base models on your dataset or we already have one Task-level preset available already ([link](https://github.com/keras-team/keras-nlp/blob/d241dd04ae60b5a0393366e816d8ac3f7ecd67f5/keras_nlp/models/gpt2/gpt2_presets.py#L130)).

There is no special API needed: the general practice is to put a `TL;DR` token between the original text and summary.

**Comment by abheesht17** [comment, marked as answer]
Just adding to @jbischof's answer here: if you are looking for summarisation, you can alternatively use BART, an encoder-decoder model (instead of GPT-2). We do have a BART preset available for summarisation (finetuned on the CNN+Daily Mail dataset). You can use it like this:

```
import keras_nlp

model = keras_nlp.models.BartSeq2SeqLM.from_preset("bart_large_en_cnn")
model.compile("greedy")
model.generate("you text goes here", max_length=100)
```

Not sure what the quality of the summaries will be, depends on your dataset, and whether it lies in the same domain as CNN+DM. If the generated summaries aren't good, you can always finetune the model.

**Comment by mattdangerw** [comment]
Yeah +1 to @abheesht17 answer, using the `"bart_large_en_cnn"` preset is the easiest way to get an article in -> summary out flow today without any training. One thing to keep in mind is that is yet to go in a pypi release, so you will want to follow instructions [here](https://github.com/keras-team/keras-nlp#installation) for getting the latest changes. Any feedback on this workflow appreciated!

Beyond that, there's a lot of other options you could consider...

- Fine-tuning a seq2seq model (such as BART) on your own dataset.
- Fine-tuning a decoder only model GPT/OPT on a summarization dataset. A common approach is to train on input text that looks like `{article_text} TL;DR {summary_text}`, which reposes the summarization problem as a single sequence problem.
- Zero-shot prompting, without fine-tuning. Today you could try that out with OPT, but the SOTA would definitely be an instruction fine-tuned LLM. We are actively working on ports on some of these models right now, so stay tuned!
